src/Serialization/Json/OkContentNormalizer.php

```diff
@@ -67,18 +67,29 @@ public function normalize($object, $format = null, array $context = []): float|i
             $uri = $this->requestStack->getMainRequest()->getUri();
             $previousPage = null;
             $nextPage = null;
+            $hasPageParameter = (bool)preg_match( '/[\?&]page=/', $uri);
 
             if ($content->hasPreviousPage()) {
                 $previousPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getPreviousPage(), $uri);
             }
             if ($content->hasNextPage()) {
-                $nextPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getNextPage(), $uri);
+                if ($hasPageParameter) {
+                    $nextPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getNextPage(), $uri);
+                } else {
+                    $nextPage = $this->addParameter('page', (string) $content->getNextPage(), $uri);
+                }
+            }
+
+            $lastPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getNbPages(), $uri);
+
+            if ($content->getCurrentPage() === 1 && $content->getNbPages() > 1 && !$hasPageParameter) {
+                $lastPage = $this->addParameter('page', (string) $content->getNbPages(), $lastPage);
```

Comment:
All of this uri parameter manipulations exerg that we need an URI management library such as  https://packagist.org/packages/league/uri here